### PR TITLE
[GEP-7] Adds check for seed provider type when changing the seed name of shoot

### PR DESCRIPTION
**Pull request by plkokanov**
**How to categorize this PR?**

/area control-plane-migration
/kind enhancement
/priority normal

**What this PR does / why we need it**:
This PR forbids changing the `Shoot.Spec.SeedName` field of a `Shoot` cluster if the new `Seed` has different cloud provider from the old one. Currently control planes cannot be migrated between `Seeds` with different providers, although we could implement this in the future.

**Which issue(s) this PR fixes**:
Part of #1631 

**Special notes for your reviewer**:

**Release note**:

```noteworthy operator
Forbid control plane migration between `Seeds` with different cloud providers.
```

### CODE REVIEW

**Review by rfranzke — DISMISSED** [review]
/lgtm

Thank you!

File: plugin/pkg/shoot/validator/admission_test.go [hunk 1]
```diff
@@ -1951,6 +1953,27 @@ var _ = Describe("validator", func() {
 
 				Expect(err).To(HaveOccurred())
 			})
+
+			It("should fail to change Seed name, because cloud provider for new Seed is not equal to cloud provider for old Seed", func() {
+				oldSeedName := fmt.Sprintf("new(%s)", seedName)
```
Comment (vpnachev) [inline plugin/pkg/shoot/validator/admission_test.go]:
```suggestion
				oldSeedName := fmt.Sprintf("old-%s", seedName)
```

**Review by vpnachev — COMMENTED** [review]
Just a minor suggestion for the name of the old seed to make it more clear. It is strange to me that the old seed is named "new..."
Also, names cannot contain brackets, ref: https://kubernetes.io/docs/concepts/overview/working-with-objects/names/#names.

File: plugin/pkg/shoot/validator/admission_test.go (hunk 1 above)
Comment (plkokanov) [reply plugin/pkg/shoot/validator/admission_test.go]: ops forgot to change this after I modified the variable name.

**Review by vpnachev — APPROVED** [review]
/lgtm

**Pull request merged**